frontend/src/hooks/useWallet.ts

```diff
@@ -0,0 +1,23 @@
+import { Web3Client } from '@/clients/web3';
+
+export function useWallet() {
+  const web3 = new Web3Client();
```

Comment:
@denishacquin great to see this, but let's add here the wrapped functions instead of just returning the web3 object. Otherwise we are not really decoupling the web3 client.
signTransaction, privateKeyToAccount, and generatePrivateKey seems better to be gotten from the useWallet